app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -131,49 +135,59 @@ public static Props props(
     }
 
     /**
-     * Ask the actor referenced by {@code ref} to start execution.
-     * <p>
-     * This will execute partition creation exactly once and then periodically thereafter using the actor's props.
+     * Ask the actor referenced by {@code ref} to stop execution.
      *
      * @param ref an {@code DailyPartitionCreator}.
      * @param timeout timeout for the operation
      * @throws ExecutionException if an exception was thrown during execution.
-     * @throws InterruptedException if the actor does not reply within the allotted timeout, or if the actor thread was
+     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
      * interrupted for other reasons.
      */
-    public static void start(
-            final ActorRef ref,
-            final Duration timeout
-    ) throws ExecutionException, InterruptedException {
-        Patterns.ask(
-                ref,
-                START_TICKING,
-                timeout
-        )
-                .thenCompose(reply -> {
-                    @SuppressWarnings("unchecked") final Optional<Throwable> o = (Optional<Throwable>) reply;
-                    final CompletableFuture<Void> future = new CompletableFuture<>();
-                    o.ifPresent(future::completeExceptionally);
-                    future.complete(null);
-                    return future;
-                })
-                .toCompletableFuture()
-                .get();
+    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
+        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
```

Comment:
Is this worth the method? Especially, since it actually has no logic specific to this actor. Moreover, you could use `PatternCS` (assuming we're on a new enough version of Akka in MP) which should simplify the call to stop (I _believe_).

Reply:
I can inline this; I had a method to be consistent with the `start` call but since that was removed there's no need for the consistency.

As far as the APIs used, PatternsCS and Patterns were merged in Akka 2.5.19 (we're running 2.5.20).

Reply:
I just looked and MAD is running Akka 2.5.16 so it makes sense that using Patterns works here but not over there.